cartographer_rviz/cartographer_rviz/drawable_submap.cc

```diff
@@ -55,16 +55,20 @@ std::string GetSubmapIdentifier(const int trajectory_id,
 }  // namespace
 
 DrawableSubmap::DrawableSubmap(const int trajectory_id, const int submap_index,
-                               Ogre::SceneManager* const scene_manager,
-                               ::rviz::Property* submap_category,
-                               const bool visible)
+                               ::rviz::DisplayContext* const context,
+                               ::rviz::Property* const submap_category,
+                               const bool visible, const float axes_length,
+                               const float axes_radius)
     : trajectory_id_(trajectory_id),
       submap_index_(submap_index),
-      scene_manager_(scene_manager),
-      scene_node_(scene_manager->getRootSceneNode()->createChildSceneNode()),
-      manual_object_(scene_manager->createManualObject(
+      context_(context),
+      scene_manager_(context->getSceneManager()),
```

Comment:
delete member? Now you have the context_, you can always get the sceneManager from it?